src/app/Document.tsx

```diff
@@ -63,11 +64,14 @@ export const Document = ({ children }: { children: ReactNode }) => {
           storageKey={COLOR_MODE_STORAGE_KEY}
         />
         <Providers>
-          <TrpcProvider>
-            <Viewport>{children}</Viewport>
-            <EnvHint />
-            <ReactQueryDevtools initialIsOpen={false} />
-          </TrpcProvider>
+          {/* Putting nuqs adapter here and not in providers as the testing requires another adapter */}
+          <NuqsAdapter>
```

Comment:
Adapters are based on Context, so you could nest them and the lowest in the tree will take precedence, if that makes things easier.